File: ported/auto/5e18ae.js
// Auto-translated from Ghidra C by tools/c-to-js/translate.js.
// Source: decompiled/c/5e18ae.c
// Edit by hand only after diff-test passes — re-running the translator will overwrite.

/** @typedef {import("../../runtime/heap.js").Heap} Heap */

import { regs } from "../../runtime/regs.js";
export function FUN_005e18ae(heap) {
  let in_AX = regs.eax & 0xffff;
  let unaff_BX = regs.ebx & 0xffff;
  let unaff_ESI = regs.esi >>> 0;
  if (((heap.i16((unaff_ESI + 0x16e)) | 0) == -1) && ((heap.u16((unaff_ESI + 0x32)) & 4) == 0)) {
    heap.setU16((unaff_ESI + 0x170), (in_AX) & 0xffff);
    heap.setU16((unaff_ESI + 0x172), (unaff_BX) & 0xffff);
    heap.setU16((unaff_ESI + 0x32), (heap.u16((unaff_ESI + 0x32)) | 8) & 0xffff);
  }
  return;
}

File: decompiled/c/5e18ae.c

void FUN_005e18ae(void)

{
  undefined2 in_AX;
  undefined2 unaff_BX;
  int unaff_ESI;
  
  if ((*(short *)(unaff_ESI + 0x16e) == -1) && ((*(ushort *)(unaff_ESI + 0x32) & 4) == 0)) {
    *(undefined2 *)(unaff_ESI + 0x170) = in_AX;
    *(undefined2 *)(unaff_ESI + 0x172) = unaff_BX;
    *(ushort *)(unaff_ESI + 0x32) = *(ushort *)(unaff_ESI + 0x32) | 8;
  }
  return;
}

